featuretools/primitives/base/primitive_base.py

```diff
@@ -51,3 +57,23 @@ def get_function(self):
 
     def get_filepath(self, filename):
         return os.path.join(config.get("primitive_data_folder"), filename)
+
+    def get_args_string(self):
+        def valid(arg):
+            error = '"{}" must be attribute of {}'
```

Comment:
can you add a docstring and/or comments this function? there's a lot of logic here that's a bit hard to quickly grasp.

it may also be helpful to actually refactor have multiple returns statements, so it's easier to follow through the different cases

Reply:
Yes, I refactored and added comments [in this commit.](https://github.com/Featuretools/featuretools/pull/481/commits/38c542d102973819a5a0ce22cbe692dad1c4a340#diff-65ffde62fc03046613de064c794ed250R61)

Reply:
looks much better! easy to follow now